universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockPredicatesEditView.java

```diff
@@ -165,19 +162,10 @@ private Tag renderPredicateForm(
         .with(renderQuestionDefinitionBox(questionDefinition))
         .with(
             div()
-                .withClasses(Styles.FLEX, Styles.FLEX_ROW, Styles.GAP_1)
-                .with(
-                    new SelectWithLabel()
-                        .setLabelText("Scalar")
-                        .setOptions(scalarOptions)
-                        .getContainer())
-                .with(
-                    new SelectWithLabel()
-                        .setLabelText("Operator")
-                        // TODO(#322): Display the right operators for the given scalar type
-                        //  (requires javascript).
-                        .setOptions(operatorOptions)
-                        .getContainer())
+                .withClasses(
+                    ReferenceClasses.PREDICATE_OPTIONS, Styles.FLEX, Styles.FLEX_ROW, Styles.GAP_1)
+                .with(createScalarDropdown(scalars))
```

Comment:
praise: Thanks for factoring a lot of this out into separate methods! Looks much better.